extension/src/popup/views/RecoverAccount/index.tsx

```diff
@@ -65,9 +66,8 @@ export const RecoverAccount = () => {
   };
 
   useEffect(() => {
-    if (publicKey) {
+    if (publicKey && publicKey !== publicKeyRef.current) {
```

Comment:
we can't just check the existence of a public key. We need to make sure it's different than the one we already have